pkg/app/master/inspectors/container/container_inspector.go

```diff
@@ -1174,6 +1176,19 @@ func (i *Inspector) initContainerChannels() error {
 		cn = i.Overrides.Network
 	}
 
+	// Top level IP info will not be populated when not using "bridge",
+	// which is only set for backwards compatibility.
+	//
+	// https://github.com/moby/moby/issues/21658#issuecomment-203527083
+	ipAddr := i.ContainerInfo.NetworkSettings.IPAddress
+	if cn != "" {
+		network, found := i.ContainerInfo.NetworkSettings.Networks[cn]
+		errutil.FailWhen(!found, fmt.Sprintf("docker-slim: error => expected NetworkSettings.Networks to contain %s: %v",
+			cn, i.ContainerInfo.NetworkSettings.Networks))
+
+		ipAddr = network.IPAddress
+	}
+
```

Comment:
good to have an `i.xc.Out.Info` call here to show what IP we ended up with

Reply:
Done.